autoscheduler/frontend/src/components/SchedulingPage/SchedulePreview/SchedulePreview.tsx

```diff
@@ -9,17 +9,37 @@ import selectSchedule from '../../../redux/actions/selectedSchedule';
 import Meeting from '../../../types/Meeting';
 import Section from '../../../types/Section';
 import * as styles from './SchedulePreview.css';
-import Grades from '../../../types/Grades';
+
+// Exported so we can test it
+export function getAverageGPATextForSchedule(schedule: Meeting[]): string {
+  let gpaSum = 0;
+  let gpaCount = 0;
+
+  schedule.forEach((meeting) => {
+    if (meeting.section.grades != null) {
+      const creditHours = meeting.section.minCredits;
+
+      // This should be weighed by the credit hours
+      gpaSum += meeting.section.grades.gpa * creditHours;
+      gpaCount += creditHours;
+    }
+  });
+
+  // Sections contained no grades
+  if (gpaCount === 0) {
+    return 'N/A';
+  }
+
+  const result = gpaSum / gpaCount;
+
+  return `${result.toFixed(2)} GPA`;
```

Comment:
I think I like formatting the GPA as `GPA: x` since when it says `N/A` it'll be more obvious what it means and makes it so the formatting will be the same whether there are grades for any sections or not. Since this is up to preference though I wonder if you guys think that looks better @gannonprudhomme @firejake308

Reply:
Yeah I agree that makes sense to me

Reply:
Yeah, that's the only thing I would change. Other than that, it's good.